controller/controller_server/utils.py

```diff
@@ -154,15 +153,42 @@ def validate_delete_snapshot_request(request):
     logger.debug("request validation finished.")
 
 
+def validate_validate_volume_capabilities_request(request):
+    logger.debug("validating validate_volume_capabilities volume request")
+
+    logger.debug("validating volume id")
+    volume_id = request.volume_id
+    if not request.volume_id:
+        raise ValidationException(messages.name_should_not_be_empty_message)
+    if config.PARAMETERS_OBJECT_ID_DELIMITER not in volume_id:
+        raise ObjectIdError(config.OBJECT_TYPE_ID_VOLUME, volume_id)
+
+    logger.debug("validating volume capabilities")
+    if not request.volume_capabilities:
+        raise ValidationException(messages.capabilities_not_set_message)
+    validate_csi_volume_capabilties(request.volume_capabilities)
+
+    logger.debug("validating secrets")
+    if request.secrets:
+        validate_secret(request.secrets)
+
+    logger.debug("request validation finished.")
+
+
+def validate_volume_context_match_volume(volume_context, volume):
+    logger.debug("validate volume_context is matching volume")
+    validated_volume_context = _get_context_from_volume(volume)
```

Comment:
this name sounds like it is already passed validation.
consider renaming to something like `context_from(_existing)?_volume`

Reply:
done